oresat_configs/__init__.py

```diff
@@ -40,47 +38,42 @@ class Card:
     """Optional child node name. Useful for CFC cards."""
 
 
-class OreSatConfig:
-    """All the configs for an OreSat mission."""
+def cards_from_csv(oresat: Consts) -> dict[str, Card]:
+    """Turns cards.csv into a dict of names->Cards, filtered by the current mission"""
 
-    CARD_CONFIG_PATHS = {
-        OreSatId.ORESAT0: ORESAT0_CARD_CONFIGS_PATH,
-        OreSatId.ORESAT0_5: ORESAT0_5_CARD_CONFIGS_PATH,
-        OreSatId.ORESAT1: ORESAT1_CARD_CONFIGS_PATH,
-    }
+    file_path = f"{os.path.dirname(os.path.abspath(__file__))}/cards.csv"
+    with open(file_path, "r") as f:
+        return {
+            row["name"]: Card(
+                row["nice_name"],
+                int(row["node_id"], 16),
+                row["processor"],
+                int(row["opd_address"], 16),
+                row["opd_always_on"].lower() == "true",
+                row["child"],
+            )
+            for row in csv.DictReader(f)
+            if row["name"] in oresat.cards_path
+        }
 
-    BEACON_CONFIG_PATHS = {
-        OreSatId.ORESAT0: ORESAT0_BEACON_CONFIG_PATH,
-        OreSatId.ORESAT0_5: ORESAT0_5_BEACON_CONFIG_PATH,
-        OreSatId.ORESAT1: ORESAT1_BEACON_CONFIG_PATH,
-    }
 
-    def __init__(self, oresat_id: OreSatId):
-        self.oresat_id = oresat_id
-        beacon_config_path = self.BEACON_CONFIG_PATHS[oresat_id]
-        beacon_config = BeaconConfig.from_yaml(beacon_config_path)
-        card_configs_path = self.CARD_CONFIG_PATHS[oresat_id]
+class OreSatConfig:
+    """All the configs for an OreSat mission."""
 
-        self.cards = {}
-        file_path = f"{os.path.dirname(os.path.abspath(__file__))}/cards.csv"
-        with open(file_path, "r") as f:
-            reader = csv.DictReader(f)
-            for row in reader:
-                name = row["name"]
-                if name in card_configs_path:
-                    del row["name"]
-                    self.cards[name] = Card(
-                        row["nice_name"],
-                        int(row["node_id"], 16),
-                        row["processor"],
-                        int(row["opd_address"], 16),
-                        row["opd_always_on"].lower() == "true",
-                        row["child"],
-                    )
+    def __init__(self, oresat: Union[OreSatId, Consts, str]):
```

Comment:
I would prefer this stay as `oresat_id` so we have the added context as to what this is.

That or add a pydoc string describing it. Either or.

Reply:
A docsting is probably the better choice here. It can't stay as an `oresat_id` because it's generally no longer just an ID, it's a `Consts`. As described in the messages for [63f8766](https://github.com/oresat/oresat-configs/pull/16/commits/63f8766959d5665f5c01e731b8f7b1bb373443a6) and [777c060](https://github.com/oresat/oresat-configs/pull/16/commits/777c0602a7f53b146c0e30a1554229c0743d0628), `Consts` has the ID, but also has other information related to a specific mission, currently long and short names, and paths to config files as well.

I kept the variable name `oresat` because then all the uses can be changed from `oresat_id` -> `oresat.id` as the `id` member is where the information now resides. If you've got ideas for the name (for both the variable and the type) I'd love to hear them though and am open to changing them. As the saying goes, the two hardest things in programming are cache invalidation, naming things, and off by one errors.